selfdrive/loggerd/logger.h

```diff
@@ -15,18 +17,39 @@ const std::string LOG_ROOT = util::getenv_default("HOME", "/.comma/media/0/reald
 
 #define LOGGER_MAX_HANDLES 16
 
+class BZFile {
+ public:
+  BZFile(const char* path) {
+    assert(file = fopen(path, "wb"));
+    int bzerror;
+    bz_file = BZ2_bzWriteOpen(&bzerror, file, 9, 0, 30);
+    assert(bzerror == BZ_OK);
+  }
+  ~BZFile() {
+    int bzerror;
+    BZ2_bzWriteClose(&bzerror, bz_file, 0, nullptr, nullptr);
+    fclose(file);
+  }
+  inline void write(void* data, size_t size) {
+    int bzerror;
+    BZ2_bzWrite(&bzerror, bz_file, data, size);
+    assert(bzerror == BZ_OK);
```

Comment:
For loggerd we probably want a more graceful handling of write errors + some cloudlog.

Reply:
Only log the first error per segment,is that ok?

Reply:
Sounds good!